aip/general/0160.md

```diff
@@ -190,9 +190,25 @@ above what is defined here. For example, a service may support the logical
 operators but only permit a certain number of them (to avoid "queries of death"
 or other performance concerns).
 
-Further structure or limitations **must** be clearly documented, **must not**
-violate requirements set forth in this document, and a non-compliant filter
-query **must** error with `INVALID_ARGUMENT`.
+Further structure or limitations **must** be clearly documented, and
+**must not** violate requirements set forth in this document.
+
+### Validation
+
+If a non-compliant or schematically invalid `filter` string is specified,
+the API **should** error with `INVALID_ARGUMENT`. Wherever validation is relaxed
+for `filter`, the API **must** document the difference.
+
+Schematic validation refers, but is not limited to, the following:
+
+- Fields referenced in the `filter` **must** exist on the filtered schema
+- Field values provided in the `filter` **must** align to the type of the field
+  - For example, for a field `int32 age` a `filter` like `"age=hello"` is
+    invalid
+- Field values for bounded data types e.g. `enum` provided in the `filter`
+  **must** be a valid value in the set
+- Field values for standardized types e.g. `Timestamp` **must** conform to the
```

Comment:
I can't remember what we decided about quoting for Timestamp, but it may be worth including here.

Reply:
I have a separate bug for that b/375437995 if you don't mind my handling it separately (though I agree they are related!)

Reply:
Yup, no problem.